admin-dev/themes/new-theme/js/components/text-to-link-rewrite-copier.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+const $ = window.$;
+
+/**
+ * Component which allows to copy regular text to url friendly text
+ *
+ * Usage example in template:
+ *
+ * <input name="source-input" class="js-link-rewrite-copier-source"> // The original text will be taken from this element
+ * <input name="destination-input" class="js-link-rewrite-copier-destination"> // Modified text will be added to this input
+ *
+ * in javascript:
+ *
+ * new TextToLinkRewriteCopier({
+ *   sourceElementSelector: '.js-link-rewrite-copier-source'
+ *   destinationElementSelector: '.js-link-rewrite-copier-destination',
+ * });
+ *
+ * If the source-input has value "test name" the link rewrite value will be "test-name".
+ * If the source-input has value "test name #$" link rewrite will be "test-name-" since #$ are un allowed characters in url.
+ *
+ * You can also pass additional options to change the event name, or encoding format:
+ *
+ * new TextToLinkRewriteCopier({
+ *   sourceElementSelector: '.js-link-rewrite-copier-source'
+ *   destinationElementSelector: '.js-link-rewrite-copier-destination',
+ *   options: {
+ *     eventName: 'change', // default is 'input'
+ *     encoding: 'US-ASCII', //default is 'UTF-8'
+ *   }
+ * });
+ *
+ */
+export default class TextToLinkRewriteCopier {
+  constructor({ sourceElementSelector, destinationElementSelector, options = { eventName: 'input', encoding: 'UTF-8' } }) {
+
+    $(document).on(options.eventName, `${sourceElementSelector}`, (event) => {
+      const $nameInput = $(event.currentTarget);
+      const langId = this._getLanguageIdByElement($nameInput);
+      let elementToModifySelector = null !== langId ? `${destinationElementSelector}[data-lang-id="${langId}"]` : destinationElementSelector;
+
+      $(elementToModifySelector).val(str2url($nameInput.val(), options.encoding));
+    })
+
+    return {};
```

Comment:
You don't need a return in constructor

Reply:
this return is used to make sure that the private functions in the js class are not reachable. For example, without it, you could easily call
```javascript
const textToLinkCopier = new TextToLinkRewriteCopier();
textToLinkCopier._getLanguageIdByElement();
```

I don't want this method to be reachable so I return empty object and this prevents from reaching this function and the result would be `undefined` in such case I think

Reply:
But this is javascript, there're no public, private or protected methods :thinking: 
If you don't want to be able to call this method, don't export it.
We'll not be able to add public method with this way of doing.

As you do for `$(document).ready`, you can add an lambda func in your constructor, more, if you don't want to be able to do something with your class, just create a function :thinking:

Reply:
done - can you check if its looks better now? :) I done this only to make it consistent across all javascript files since this return was used quite often and the classes there used instead of functions